packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectCompanion.kt

```diff
@@ -31,7 +32,7 @@ public interface RealmObjectCompanion {
     public val `io_realm_kotlin_className`: String
     public val `io_realm_kotlin_fields`: Map<String, KProperty1<BaseRealmObject, Any?>>
     public val `io_realm_kotlin_primaryKey`: KMutableProperty1<*, *>?
-    public val `io_realm_kotlin_isEmbedded`: Boolean
+    public val `io_realm_kotlin_classKind`: RealmClassKind
```

Comment:
Why do we expose the `classKind` here if we have it available in `io_realm_kotlin_schema`() via `RealmClassImpl`?

Reply:
To be honest I'm not 100% sure 🙈 , I assume it makes some checks a lot faster rather than going through a bunch of collection lookups.

Reply:
If it is a duplicate then I think we should be a single implementation in `ClassImpl, it would be easier to maintain as it does not require injecting the attribute from the compiler plugin.

Reply:
The companion object interface and the schema information has evolved quite a bit since we initially laid out the structure, so we could maybe clean it up and/or tweak it for the various use cases. I would be in favor of putting the static information in the companion. But anyway I don't think that it is something to address in this PR.